components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -248,7 +248,14 @@ interface AssetRatioController {
   GetGasOracle() => (GasEstimation1559? estimation);
 };
 
+struct SwapConfig {
+  string swap_api_url;
+  string buy_token_percantage_fee;
+  string fee_recipient;
+};
+
 interface SwapController {
+  GetSwapConfiguration(string chain_id) => (SwapConfig config);
```

Comment:
Created a desktop issue to use that function https://github.com/brave/brave-browser/issues/18546

Reply:
Maybe a return result of `SwapConfig?` here so absl::optional<mojom::SwapConfig>` so that you can return an optional result when there is no known swap config for the passed in chain.